docs/website/docs/development/architecture/how-odo-works.md

```diff
@@ -0,0 +1,1382 @@
+|-------------------------------------|---------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `service.binding/backend_ip`        | `path={.spec.clusterIP}`                                                                          | exposes the Service `clusterIP` address as binding data, so this can be used as a backing service via the Service Binding Operator (SBO). More details on [SBO documentation](https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/adding-annotation.html) |
+| `service.binding/backend_port`      | `path={.spec.ports},`<br/>`elementType=sliceOfMaps,`<br/>`sourceKey=name,`<br/>`sourceValue=port` | exposes the Service ports as binding data, so this can be used as a backing service via the Service Binding Operator (SBO). More details on [SBO documentation](https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/adding-annotation.html)               |
+
+Note that any additional annotations defined via the `components[].container.annotation.service` Devfile field will also be added to this resource.
+
+
+<details>
+<summary>Example</summary>
+
+<table>
+<thead>
+<tr>
+<td>Devfile</td>
+<td></td>
+<td>Kubernetes Service</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+
+```yaml
+metadata:
+  # highlight-start
+  name: my-sample-java-springboot
+  projectType: spring
+  # highlight-end
+  language: java
+components:
+- name: tools
+  container:
+     annotation:
+        service:
+           # highlight-start
+           example.com/my-svc-annotation: value-1
+           # highlight-end
+```
+</td>
+<td> => </td>
+<td>
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+   # highlight-next-line
+   name: my-sample-java-springboot-app
+   annotations:
+      # highlight-start
+      service.binding/backend_ip: path={.spec.clusterIP}
+      service.binding/backend_port: path={.spec.ports},elementType=sliceOfMaps,sourceKey=name,sourceValue=port
+      example.com/my-svc-annotation: value-1
+      # highlight-end
+spec:
+   type: ClusterIP
+   selector:
+      # highlight-next-line
+      component: my-sample-java-springboot-app
+```
+</td>
+</tr>
+</tbody>
+</table>
+
+</details>
+
+
+#### Labels
+
+By default, `odo` adds the following labels to the Service:
+
+| Key                                    | Description                                                                                                                                          | Example Value       |
```

Comment:
These labels are the same as Deployments, no? Perhaps you don't need to redefine them and simply point to that section?

Reply:
Yup, they are the same. I'll list them under a dedicated section.